ui/redux/selectors/claims.js

```diff
@@ -936,7 +952,37 @@ export const selectTakeOverAmountForName = (state: State, name: string) => {
 export const selectMyPurchasedClaims = createSelector(selectState, (state) => state.myPurchasedClaims || []);
 
 export const selectPurchaseMadeForClaimId = (state: State, claimId: string) => {
-  const purchasedClaimIds = selectMyPurchasedClaims(state);
+  const purchasedClaims = selectMyPurchasedClaims(state);
+
+  let matchedClaimIds = [];
+  for (const purchasedClaim of purchasedClaims) {
+    matchedClaimIds.push(purchasedClaim.reference_claim_id);
+    matchedClaimIds.push(purchasedClaim.target_claim_id);
+  }
+
+  return (matchedClaimIds.length && matchedClaimIds.includes(claimId)) || false;
+};
+
+export const selectPurchaseForClaimId = (state: State, claimId: string) => {
+  const purchasedClaims = selectMyPurchasedClaims(state);
+
+  const purchasedClaimsByClaimId = purchasedClaims.filter((purchase) => {
+    return purchase.reference_claim_id === claimId || purchase.target_claim_id === claimId;
+  });
+
+  return purchasedClaimsByClaimId;
+};
```

Comment:
:small_red_triangle: Memoize this

This selector returns a non-trivial value (or unstable reference), i.e. array/object, which will then cause the component that uses it to re-render on each state change.

But it doesn't seem like this selector is used.  Delete?